Task server backend

athithyaaselvam opened the pull request:
## What changes were proposed in this pull request?

- Task server backend changes

## How was this patch tested?

##- (Please explain how this patch was tested. Ex: unit tests, manual tests)
##- (If this patch involves UI changes, please attach a screen-shot; otherwise, remove this)

amitsrivastava commented on apps/filebrowser/src/filebrowser/tasks.py:
```diff
@@ -0,0 +1,186 @@
+
+    return
+
+import os
+import shutil
+import psutil
+from datetime import datetime, timedelta
```
Can possibly move these imports to the top.

amitsrivastava commented on apps/filebrowser/src/filebrowser/views.py:
```diff
@@ -1458,10 +1458,11 @@ def _create_response(request, _fs, result="success", data="Success"):
   }
 
-def perform_upload(request, *args, **kwargs):
+def perform_upload_task(request, *args, **kwargs):
```
Why rename the function? It seems that it still supports the non task server method.

amitsrivastava commented on desktop/core/src/desktop/api2.py:
```diff
@@ -688,6 +692,155 @@ def share_document(request):
+    return JsonResponse({'error': str(e)}, status=500)
+
+  if task_name == 'document cleanup':
+    keep_days = task_params.get('keep-days')
```
add a default value too.

amitsrivastava commented on desktop/core/src/desktop/api2.py:
```diff
@@ -688,6 +692,155 @@ def share_document(request):
+  except json.JSONDecodeError as e:
+    return JsonResponse({'error': str(e)}, status=500)
+
+  if task_name == 'document cleanup':
```
I would recommend creating all the task names as constants in tasks.py and skip the spaces in values.
Something like
```
DOCUMENT_CLEANUP = 'DOCUMENT_CLEANUP'
```

amitsrivastava commented on desktop/core/src/desktop/api2.py:
```diff
@@ -688,6 +692,155 @@ def share_document(request):
+
+  return JsonResponse({'isFinalized': is_finalized, 'isRunning': is_running, 'isFailure': is_failure, 'isRevoked': is_revoked})
+
+from django.http import JsonResponse
+from celery.app.control import Control
+from desktop.celery import app as celery_app
```
Is there a reason to keep these imports here in between the file? Unless these is some dependency issue, the imports should be at the top. It helps with maintaining the imports.

amitsrivastava commented on desktop/core/src/desktop/celery.py:
```diff
@@ -25,13 +25,38 @@
 os.environ.setdefault('DJANGO_SETTINGS_MODULE', 'desktop.settings')
 
-app = Celery('desktop')
+class MyCelery(Celery):
```
Why `MyCelery`? Why not `HueCelery`?

athithyaaselvam commented:
Moved the imports to the top and renamed MyCelery. We have split the PR into two parts and abandoning this PR.
1. [Adding Task server backend changes for Hue #3687](https://github.com/cloudera/hue/pull/3687)  
2. [Task server Docker changes and backend tasks changes #3688](https://github.com/cloudera/hue/pull/3688  )
